uiowa/uiowa
# Mega PR: WYSIWYG Buttons Refactor, Admissions card link updates, Table updates

Resolves #3078 
Relates to https://github.com/uiowa/uiowa/issues/260
Relates to https://github.com/uiowa/uiowa/issues/2532
Resolves #3262 
Resolves #3330 
Resolves #3395

- Removes our custom LinkIt solution for adding button classes to WYSIWYG links and replaces it with style dropdown options on the `<a>` tag.
- On all sites but HR, loops through text_long and text_with_summary type fields along with our ignored view config and replaces bootstrap button classes with uids classes.
- Does the same thing to remove table classes.
- Add bold and light intro classes to style dropdown on the `<p>` tag.
- Splits the filtered_html text format for HR which has different classes for the lead and buttons.

# To Test

To make sure that you get the UIDS update, follow these steps:
```
rm -rf ./node_modules
yarn cache clean
yarn install
yarn workspace uids_base gulp --development
yarn workspace admissions_core build
```

## Button updates

- Sync `sandbox` `hr` and other sites (e.g. a website with separate content type with text_long or text_with_summary fields).
- See https://sandbox.local.drupal.uiowa.edu/bootstrap-buttons. All of the markup has been changed from btn* classes to bttn* classes. Revert to a previous revision. You should still see the bttn* markup.
- Make changes to the content to confirm the markup stays and that there are no weird errors.
- See https://sandbox.local.drupal.uiowa.edu/sandbox-people. See the header content buttons are bttn* markup.
- See https://hr.local.drupal.uiowa.edu/advanced-layout-guide-not-publication and see that the buttons on the page are still the HR bootstrap based styles. Edit the page and see that Lead and Button style options still use bootstrap classes.

### Go the extra mile

- Query the entire database to see if `btn-` exists. It shouldn't except maybe for a key_value_expire table entry.

## Area of Study (Admissions) card link updates

- `blt frontend`
- Edit http://admissions.local.drupal.uiowa.edu/academics/accounting and add links to to the Admissions Requirement link fields.  Save the page and verify that the link button works but the entire card is not linked. 
- Edit the page again and add a "Story" to the page and verify that the story card is still clickable. 

## Print links (Admissions) permissions updates
- Masquerade as a webmaster and go to https://admissions.local.drupal.uiowa.edu/areas-of-study/print-links.   Try clicking on a pdf link in the "Tests" table at the bottom of the page.   You should **not** be seeing an "access denied" message. 

## Table updates
- Add a large table to every layout region and verify that the table shows the `overflow-y` scrollbar:
<details>

```
<table>
	<caption>Tables can have captions now.</caption>
	<thead>
		<tr>
			<th>Person</th>
			<th>Number</th>
			<th>Third Column</th>
			<th>Fourth Column</th>
			<th>Fifth Column</th>
			<th>Sixth Column</th>
			<th>Seventhth Column</th>
			<th>Eighth Column</th>
			<th>Ninth Column</th>
			<th>Tenth Column</th>
			<th>Eleventh Column</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<th>Someone Lastname</th>
			<td>900</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th><a href="#">Person Name</a></th>
			<td>1200</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th>Another Person</th>
			<td>1500</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th>some other person</th>
			<td>2800</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th>Someone and a Lastname</th>
			<td>500</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th><a href="#">Linked Name</a></th>
			<td>1720</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th>Another random person</th>
			<td>1200</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th>Some other Person</th>
			<td>2200</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th>Firstname and something</th>
			<td>920</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th><a href="#">Linked Other Name</a></th>
			<td>1208</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th>Another Person Randomly selected</th>
			<td>1550</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
		<tr>
			<th>Last One</th>
			<td>2830</td>
			<td>9999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
			<td>1999</td>
		</tr>
	</tbody>
</table>
```

</details>
